server/punishments.js

```diff
@@ -428,7 +449,7 @@ const Punishments = new (class {
 	 * @return {string}
 	 */
 	renderEntry(entry, id) {
-		let row = [entry.punishType, id, entry.keys.join(','), ...entry.rest];
+		let row = [entry.punishType, id, [entry.ips.join(','), entry.userids.join(',')].filter(el => el), entry.expireTime, entry.reason, ...entry.rest];
```

Comment:
This converts an explicit cast into an implicit cast, and only works because the default behavior of implicitly casting an array to a string is to join by `,`.

```suggestion
		const keys = entry.ips.concat(entry.userids);
		let row = [entry.punishType, id, keys.join(','), entry.expireTime, entry.reason, ...entry.rest];
```